statefun-e2e-tests/statefun-smoke-e2e-java/src/main/java/org/apache/flink/statefun/e2e/smoke/CommandInterpreterFn.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.statefun.e2e.smoke;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.flink.statefun.sdk.java.*;
+import org.apache.flink.statefun.sdk.java.message.Message;
+
+public class CommandInterpreterFn implements StatefulFunction {
+
+  static final TypeName TYPENAME = TypeName.typeNameOf("statefun.e2e", "interpreter");
+  static final ValueSpec<Long> state = ValueSpec.named("state").withLongType();
+  static final StatefulFunctionSpec SPEC =
+      StatefulFunctionSpec.builder(TYPENAME).withSupplier(CommandInterpreterFn::new).build();
+  final CommandInterpreter interpreter;
+
+  public CommandInterpreterFn() {
+    int numOfInstances = 2; // TODO: make this configurable.
```

Comment:
Not sure how to pass the config in...

Reply:
One thing we can consider, is to move the `SPEC` into `CommandInterpreterAppServer`, and pass in this config through the Java program main arguments:

```
class AppServer {

    public static void main(String[] args) {
        final int numInterpreterInstances = parseArgs(args);
        final CommandInterpreter interpreter = new CommandInterpreter(numInterpreterInstances);
        
        final StatefulFunctions fns = new StatefulFunctions();
        fns.withStatefulFunction(StatefulFunctionSpec.builder(...)
            .withSupplier(() -> new CommandInterpreterFn(interpreter)).build());
    }
}
```

Then we should probably revert the changes to the constructor of `CommandInterpreterFn` to accept the interpreter as a dependency.